### PR TITLE
docs: add design doc about the new aggregate framework

### PR DESCRIPTION
**Pull request by zz-jason**
## What have you changed? (mandatory)

add design doc about the new aggregate framework.

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Does this PR need to be added to the release notes? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/tidb/issues/6952

### CODE REVIEW

**Comment by zz-jason** [comment]
@shenli @lilin90 PTAL

File: docs/design/2018-07-01-refactor-aggregate-framework.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Proposal: A new aggregate function execution framework
+
+- Author(s):     [@zz-jason](https://github.com/zz-jason)
+- Last updated:  2017-07-01
```
Comment (coocood) [inline docs/design/2018-07-01-refactor-aggregate-framework.md]: 2018

File: docs/design/2018-07-01-refactor-aggregate-framework.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Implement the `AVG` function under the new framework. [@zz-jason](https://github.com/zz-jason)
+- Implement other aggregate functions under the new framework. [@XuHuaiyu](https://github.com/XuHuaiyu)
+- Add more integration tests using the random query generator. [@XuHuaiyu](https://github.com/XuHuaiyu)
+- Add unit tests for the new framework. [@XuHuaiyu](https://github.com/XuHuaiyu)
+- Infer input parameter type for new aggregate functions when extracting a `Project` operator under the `Aggregate` operator. [@XuHuaiyu](https://github.com/XuHuaiyu)
+- Use the new framework in the hash aggregate operator. [@XuHuaiyu](https://github.com/XuHuaiyu)
+
+## Open issues (if applicable)
+
+- https://github.com/pingcap/tidb/issues/7042: add unit tests for package "aggfuncs".
```
Comment (coocood) [inline docs/design/2018-07-01-refactor-aggregate-framework.md]: According to the template, `Open issue` is for the issues that we don't have a solution.

**Comment by zz-jason** [comment]
@coocood done, PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**